File: gui_apps/Doodle/step02/DrawWindow.cc
// DrawWindow.cc: implements DrawWindow class
#include "DrawWindow.h"
#include <QMessageBox>
#include <QtGui>

DrawWindow::DrawWindow()
{
   setWindowTitle("Qt5 Doodle - Step02: Handling Events");
   // setStyleSheet("background-color: white;");
   resize(QGuiApplication::primaryScreen()->availableSize() * 4 / 5);
}

void DrawWindow::closeEvent(QCloseEvent *event)
{
   // window is about to close, prompt user & decide if ok to quit
   // based on user's response.
   qDebug() << "DrawWindow::closeEvent() called!" << Qt::endl;
   auto ret = QMessageBox::question(this,
       tr("Qt Scribble Tutorial"),
       tr("This will close the application.\nOk to quit now?"),
       QMessageBox::Yes | QMessageBox::No,
       QMessageBox::No);

   switch (ret) {
   case QMessageBox::Yes:
       // ok to quit
       event->accept();
       break;
   default:
       // don't quit yet
       event->ignore();
   }
}

void DrawWindow::mousePressEvent(QMouseEvent *event)
{
    if (event->button() == Qt::LeftButton) {
        QMessageBox::information(this, tr("Qt Scribble Tutorial"), tr("You have pressed the LEFT mouse button!"));
    } else if (event->button() == Qt::RightButton) {
        QMessageBox::information(this, tr("Qt Scribble Tutorial"), tr("You have pressed the RIGHT mouse button!"));
    }
}

File: gui_apps/PyDoodle/step02/drawWindow.py
"""
// ============================================================================
// drawWindow.py: custom QMainWindow derived class for main window
//
// Tutorial - Qt Scribble Application
// Based on a similar tutorial for Borland ObjectWindows Library (OWL)
// Created by Manish Bhobe.
// My experiments with the Qt Framework. Use at your own risk!!
// ============================================================================
"""
import sys
from PyQt5.QtCore import *
from PyQt5.QtGui import *
from PyQt5.QtWidgets import *

class DrawWindow(QMainWindow):
    def __init__(self, *args, **kwargs):
        super(QMainWindow, self).__init__(*args, **kwargs)
        self.setWindowTitle("PyQt5 Doodle - Step02: Handling Events")
        self.setStyleSheet("background-color: white")
        self.setGeometry(QRect(100,100,640,480))

    # operating system Events
    def closeEvent(self, e: QCloseEvent) -> None:
        resp = QMessageBox.question(self, "Confirm Close",
                                    "This will close the application.\nOk to quit?",
                                    QMessageBox.Yes | QMessageBox.No, QMessageBox.No)
        if resp == QMessageBox.Yes:
            e.accept()
        else:
            e.ignore()

    def mousePressEvent(self, e: QMouseEvent) -> None:
        if e.button() == Qt.LeftButton:
            QMessageBox.information(self, "PyQt Doodle",
                                    "You have pressed the LEFT mouse button")
        elif e.button() == Qt.RightButton:
            QMessageBox.information(self, "PyQt Doodle",
                                    "You have pressed the RIGHT mouse button")
